mongo/client_encryption.go

```diff
@@ -42,6 +43,40 @@ func NewClientEncryption(keyVaultClient *Client, opts ...*options.ClientEncrypti
 	db, coll := splitNamespace(ceo.KeyVaultNamespace)
 	ce.keyVaultColl = ce.keyVaultClient.Database(db).Collection(coll, keyVaultCollOpts)
 
+	credentialCallback := func() interface{} {
+		// By default, do nothing.
+		return nil
+	}
+
+	awsEmpty := false
+	if awsMap, ok := ceo.KmsProviders["aws"]; ok {
+		awsEmpty = true
+		for range awsMap {
+			awsEmpty = false
+			break
+		}
+	}
+
+	// If an empty "aws" map is configured, the driver is responsible for fetching "aws" credentials.
+	if awsEmpty {
+		// Remove the "aws" field from the map. libmongocrypt considers an empty "aws": {} an error.
+		delete(ceo.KmsProviders, "aws")
+		if ceo.CredentialCallback != nil {
+			credentialCallback = ceo.CredentialCallback
+		} else {
+			// Use callback to fetch credentials using same method as MONGODB-AWS.
+			credentialCallback = func() interface{} {
```

Comment:
Should we future-proof the API so that we can support callbacks for multiple KMS providers? I think that would require some way to find out from libmongocrypt which provider it's requesting credentials for.